### PR TITLE
Feature/136 password recovery back end functionality implementation

### PR DESCRIPTION
**Pull request by davideastmond**
Resolves: https://github.com/chingu-voyages/v25-bears-team-05/issues/136
- This PR:
- Creates route that implements validation of requests and updating of user's password
- Updated some tests and adjusted some functions
- Random linting fixes.

### CODE REVIEW

File: src/middleware/password-recovery/validate-id-data.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * @param next
+ */
+export async function validateIdDataRequest(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  const { id, data } = req.query;
+
+  if (req.query.devBypass && !!req.query.devBypass === true) {
```
Comment (lily-law) [inline src/middleware/password-recovery/validate-id-data.ts:25]: Maybe we should check that we're not in production here with `const isProduction = process.env.NODE_ENV && process.env.NODE_ENV.match("production");`

`(req.query.devBypass && !!req.query.devBypass === true)` seems a little redundant. `if (value)` already infers if it's any value other then falsy (""||0||false||undefined||null) then it's truthy. And !!req.query.devBypass casts the truthy or falsy to a hard true or false value.
Normally if we're wanting to check that a particular value is set as true on the query we could do a strict comparison between strings, value === "true", or if we only care if the value is not a falsy just the value it's self is sufficient.

File: src/middleware/password-recovery/validate-id-data.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    console.warn(
+      "\x1b[31m",
```
Comment (lily-law) [inline src/middleware/password-recovery/validate-id-data.ts:27]: :1st_place_medal: nice I need to start using colours more.

**Review by lily-law — APPROVED** [review]
Looking good, nice and clear coding :+1:

File: src/middleware/password-recovery/validate-id-data.ts (hunk 1 above)
Comment (davideastmond) [reply src/middleware/password-recovery/validate-id-data.ts]: Hey. Actually I actually deleted this code (but forgot to push my changes). 
This code was just a workaround to get the sever to respond a certain way so I could get the front end UI to render.

**Pull request merged**